backend.go

```diff
@@ -5,45 +5,78 @@ package fakemachine
 
 import (
 	"fmt"
+	"sort"
 )
 
-// A list of backends which are implemented
+// Main list of backends vs. their order in the "auto" algoritmn.
+func implementedBackends(m *Machine) map[int]backend {
+	return map[int]backend{
+		10: newKvmBackend(m),
+		20: newUmlBackend(m),
+		30: newQemuBackend(m),
+	}
```

Comment:
any reason for this being a map. Judging by the remainder of the code a simple array in priority order seems more suilted?

Reply:
i will rework it to an array ;-)